packages/@aws-cdk-testing/framework-integ/test/aws-cloudfront-origins/test/integ.function-url-origin-oac.ts

```diff
@@ -0,0 +1,82 @@
+import * as lambda from 'aws-cdk-lib/aws-lambda';
+import * as cloudfront from 'aws-cdk-lib/aws-cloudfront';
+import * as cdk from 'aws-cdk-lib';
+import { ExpectedResult, IntegTest, Match } from '@aws-cdk/integ-tests-alpha';
+import * as origins from 'aws-cdk-lib/aws-cloudfront-origins';
+
+const app = new cdk.App();
+
+const stack = new cdk.Stack(app, 'integ-cloudfront-function-url-origin-oac');
+
+// Create the Lambda function with a custom response
+const fn = new lambda.Function(stack, 'MyFunction', {
+  code: lambda.Code.fromInline(`
+    exports.handler = async () => ({
+      statusCode: 200,
+      body: 'Hello!!'
+    });
+  `),
+  handler: 'index.handler',
+  runtime: lambda.Runtime.NODEJS_20_X,
+});
+
+// Add a Lambda Function URL
+const fnUrl = fn.addFunctionUrl({
+  authType: lambda.FunctionUrlAuthType.AWS_IAM,
+});
+
+// Create CloudFront Distribution with OAC
+const distribution = new cloudfront.Distribution(stack, 'Distribution', {
+  defaultBehavior: {
+    origin: origins.FunctionUrlOrigin.withOriginAccessControl(fnUrl),
```

Comment:
Also I would recommend to have a another integ test for the custom OAC with new L2 construct `FunctionUrlOriginAccessControl` and using it in `withOriginAccessControl` method

Reply:
fixed,